src/core/uploadFrameMediaToIPFS.ts

```diff
@@ -0,0 +1,50 @@
+import fs from 'fs';
+
+type FrameMediaOptions = {
+  pinataJwt: string;
+  gatewayUrl: string;
+  pathToMedia: string;
+};
+
+/**
+ * Take a path string, read the file, and upload it to IPFS
+ *
+ * If the path is valid and upload is successful, return the IPFS gateway URL for use in the frame, otherwise, return an error.
+ *
+ * @param mediaOptions The JSON that includes the path string, the Pinata JWT, and the Gateway URL to use.
+ */
+async function uploadFrameMediaToIPFS(
```

Comment:
@cnasc @Yuripetusko what's your take on this utility?

I am reflecting if it's worth to have something that helps in general upload things to IPFS.

Reply:
I think it's good, my only comment would be that there should be options for other providers (filebase, nftstorage, etc)

Reply:
Ah actually just noticed that this is part of onchainkit. This should always be a backend code due to api key exposure and formadata handling, I am wondering if `build-onchain-apps` is a better place, and treat it more like an example to copy